src/core/net/netif.cpp

```diff
@@ -325,6 +327,97 @@ ThreadError Netif::RemoveUnicastAddress(const NetifUnicastAddress &aAddress)
     return error;
 }
 
+
+ThreadError Netif::AddExternalUnicastAddress(const NetifUnicastAddress &aAddress)
+{
+    ThreadError error = kThreadError_None;
+    int8_t index = 0;
+
+    for (NetifUnicastAddress *cur = mUnicastAddresses; cur; cur = cur->GetNext())
+    {
+        if (memcmp(&cur->mAddress, &aAddress.mAddress, sizeof(otIp6Address)) == 0)
+        {
+            VerifyOrExit(GetExtUnicastAddressIndex(cur) != -1, error = kThreadError_InvalidArgs);
+
+            cur->mPreferredLifetime = aAddress.mPreferredLifetime;
+            cur->mValidLifetime = aAddress.mValidLifetime;
+            cur->mPrefixLength = aAddress.mPrefixLength;
+            ExitNow(error = kThreadError_Busy);
```

Comment:
I've been thinking about this error code a bit.  I'm not sure `kThreadError_Busy` is best here.  When we've reached this location, we've successfully updated the IPv6 address as requested.  I'm inclined to say that this should return `kThreadError_None` such that helpers like `SuccessOrExit()` work as expected.  If we return something other than `kThreadError_None`, then I'd rather not update any state.